src/pages/index.js

```diff
@@ -0,0 +1,375 @@
+  .catch((error) => {
+    console.error(error);
+  });
+
+// Profile Elements
+const profileName = document.querySelector(".profile__name");
+const profileDescription = document.querySelector(".profile__description");
+const profileAvatar = document.querySelector(".profile__avatar");
+
+// Modal Input Elements
+const editModalNameInput = document.querySelector("#profile-name-input");
+const editModalDescriptionInput = document.querySelector(
+  "#profile-description-input"
+);
+const postModalLinkInput = document.querySelector("#image-input");
+const postModalNameInput = document.querySelector("#caption-input");
+const editAvatarInput = document.querySelector("#avatar-input");
+
+// Form and Modal Elements
+const editFormElement = document.querySelector("#edit-modal-form");
+const postFormElement = document.querySelector("#post-modal-form");
+const editAvatarFormElement = document.querySelector("#avatar-modal-form");
+const editModal = document.querySelector("#edit-modal");
+const postModal = document.querySelector("#post-modal");
+const viewImageModal = document.querySelector("#view-image-modal");
+const modalImage = viewImageModal?.querySelector(".modal__image");
+const modalCaption = viewImageModal?.querySelector(".modal__caption");
+const editAvatarModal = document.querySelector("#edit-avatar");
+const deleteImageModal = document.querySelector("#delete-image");
+
+// Button Elements
+const postButton = document.querySelector(".profile__post-btn");
+const settingsButton = document.querySelector(".profile__settings");
+const editAvatarButton = document.querySelector(".profile__avatar-btn");
+const editModalExitButton = document.querySelector("#edit-modal-close-btn");
+const postModalExitButton = document.querySelector("#post-modal-close-btn");
+const viewImageModalCloseButton = document.querySelector(
+  "#view-image-modal-close-btn"
+);
+const postSubmitBtn = postModal.querySelector(".modal__submit-btn");
+const editAvatarExitButton = document.querySelector("#edit-avatar-close-btn");
+const deleteImageExitButton = document.querySelector(
+  "#close-btn-modal-close-btn"
+);
+const deleteImageDeleteButton = document.querySelector("#confirm-delete-btn");
+const deleteImageCancelButton = document.querySelector("#cancel-delete-btn");
+const editProfileSubmitButton = document.querySelector(
+  "#edit-profile-submit-btn"
+);
+
+// Card Template and List Elements
+const cardTemplate = document.querySelector("#card-template");
+const cardList = document.querySelector(".cards__list");
+
+// Event Listeners
+if (settingsButton) {
+  settingsButton.addEventListener("click", () => {
+    editModalNameInput.value = profileName.textContent;
+    editModalDescriptionInput.value = profileDescription.textContent;
+    resetValidation(
+      editFormElement,
+      [editModalNameInput, editModalDescriptionInput],
+      settings
+    );
+    openModal(editModal);
+  });
+}
+if (editModalExitButton) {
+  editModalExitButton.addEventListener("click", () => {
```

Comment:
**Could be improved**

You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.